share/spice/forvo/forvo.js

```diff
@@ -12,143 +12,34 @@ function ddg_spice_forvo (api_result) {
     var script = $('[src*="/js/spice/forvo/"]')[0];
     var source = $(script).attr("src");
     var query = source.match(/forvo\/([^\/]+)\/\w+/)[1];
-
+    query = decodeURIComponent(query);
+    
     // Display the Spice plug-in.
     Spice.add({
-        data             : api_result,
-        name             : "Pronunciation",
-        sourceUrl       : "http://www.forvo.com/search/" + query,
-        sourceName      : "Forvo",
-        id              : "forvo",
-        template_frame   : "list",
-        templates : {
-            items         : api_result.items, //list,
-            item: Spice.forvo.forvo,
-            show          : 3,
-            max           : 5,
-            type          : "ul",
-            use_alternate_template: false
+        id: "forvo",
+        name: "Pronunciation",
+        data: {
+            list: api_result.items
         },
-        
-        
-    });
-
-    // This gets called when the sound is finished playing
-    // or when another player is playing. It basically just resets the look of the player.
-    var clearPlayer = function() {
-        var li = $(".spice2list_item");
-        li.removeClass("sm2_playing");
-        li.removeClass("sm2_paused");
-        li.addClass("sm2_stopped");
-        soundManager.stopAll();
-    };
-
-    // Loads and plays the audio file.
-    var isFailed = false;
-    var playSound = function(anchor) {
-        soundManager.stopAll();
-
-        if(isFailed) {
-            setTimeout(function() {
-                clearPlayer();
-            }, 1000);
-        } else {
-            var sound = soundManager.createSound({
-                id: anchor.attr("id"),
-                url: anchor.data("stream"),
-                onfinish: function() {
-                    clearPlayer();
-                }
-            });
-
-            sound.play();
-        }
-    };
-
-    // Initialize SoundManager2.
-    window.SM2_DEFER = true;
```

Comment:
I'm excited to see how much smaller this file is now that we've abstracted away the SoundManager code!

One thing I'd like to mention, we've changed the way we organize these callback files slightly: Now we wrap all the code in a closure, so we can safely create global variables/functions outside out callback function, yet prevent them from polluting the global namespace.

Can you like to re-sructure this file to mimic the canonical shape? Here's a good, simple example: https://github.com/duckduckgo/duckduckgo-template-helpers/blob/master/template_helpers.js#L11

(BTW, if you use `duckpan new` it will create all the files for you, which is ideal when building a new IA)

In order to get the audio working, we'll need to wrap the `Spice.add` call inside `DDG.require()`:

```js
DDG.require('audio', function(){
    Spice.add({
        // Display properties go here
    });
});
```